### PR TITLE
Nt 111 create view user mention

### PR DESCRIPTION
**Pull request by fadestocodes**
## JIRA Ticket Link
https://atriadev.atlassian.net/browse/NT-111
## Migrations Required?

NO :negative_squared_cross_mark:

## Summary (a few sentences describing what this PR is doing)
Created a url path and View method to use the request query param to call the appropriate Service method. The view serializes the data from the Service method to pass in the Response object. I also wrote a unit test for the endpoint to test different cases.
## Checks

- [x] I have ran all the tests locally, and they all pass

## Impacted Areas in Application

List the areas in the app this PR is going to affect

- general backend

## Learnings

- better understanding of serializing data in Django in the DRF workflow

### CODE REVIEW

**Review by colecathcart — APPROVED** [review]
Looks good

File: townhall/users/views.py [hunk 1]
```diff
@@ -285,3 +285,23 @@ def update_user(self, request, user_id):
             # Print the error for debugging
             print(f"Unexpected error in update_user: {e}")
             print(f"Error type: {type(e).__name__}")
+
+    # SEARCH USERS TO MENTION
+    @action(detail=False, methods=["get"], url_path="mention")
+    def mention_user(self, request):
+
+        query = request.query_params.get("query", "")
```
Comment (ansel-rangers11) [inline townhall/users/views.py:293]: Add test case for this empty stiring if no query is provided

Comment (fadestocodes) [reply townhall/users/views.py:293]: done!
